libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -0,0 +1,386 @@
+"""Wrapper around Google VertexAI chat-based models."""
+from __future__ import annotations
+
+import base64
+import logging
+import re
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Any, Dict, Iterator, List, Optional, Union, cast
+from urllib.parse import urlparse
+
+import requests
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain_core.language_models.chat_models import (
+    BaseChatModel,
+    generate_from_stream,
+)
+from langchain_core.messages import (
+    AIMessage,
+    AIMessageChunk,
+    BaseMessage,
+    HumanMessage,
+    SystemMessage,
+)
+from langchain_core.outputs import ChatGeneration, ChatGenerationChunk, ChatResult
+from langchain_core.pydantic_v1 import root_validator
+
+from langchain_google_vertexai.llms import (
+    _VertexAICommon,
+    is_codey_model,
+    is_gemini_model,
+)
+from langchain_google_vertexai.utils import (
+    load_image_from_gcs,
+    raise_vertex_import_error,
+)
+
+if TYPE_CHECKING:
+    from vertexai.language_models import (
```

Comment:
if we want, we can make this required now?

Reply:
done